bioptim/dynamics/configure_problem.py

```diff
@@ -140,13 +140,34 @@ def torque_driven(
         ConfigureProblem.configure_qdot(nlp, True, False)
         ConfigureProblem.configure_tau(nlp, False, True, fatigue)
 
-        if implicit_dynamics:
+        if multibody_dynamics == MultiBodyDynamics.SEMI_EXPLICIT:
             ConfigureProblem.configure_qddot(nlp, False, True)
+            ocp.implicit_constraints.add(
```

Comment:
Also, it feels weird that "implicit_constraints" stands for "semi_implicit" too. I do not condone using an extra list though and cannot think of a better name...